modules/modify-vm-template/pkg/utils/parse/clioptions.go

```diff
@@ -0,0 +1,269 @@
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apimachinery/pkg/api/resource"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+)
+
+const (
+	templateNameOptionName      = "template-name"
+	templateNamespaceOptionName = "template-namespace"
+	colonSeparator              = ":"
+)
+
+type CLIOptions struct {
+	TemplateName        string            `arg:"--template-name,env:TEMPLATE_NAME,required" placeholder:"NAME" help:"Name of a template"`
+	TemplateNamespace   string            `arg:"--template-namespace,env:TEMPLATE_NAMESPACE" placeholder:"NAMESPACE" help:"Namespace of a template"`
+	CPUSockets          string            `arg:"--cpu-sockets,env:CPU_SOCKETS" placeholder:"CPU_SOCKETS" help:"Number of CPU sockets"`
+	CPUCores            string            `arg:"--cpu-cores,env:CPU_CORES" placeholder:"CPU_CORES" help:"Number of CPU cores"`
+	CPUThreads          string            `arg:"--cpu-threads,env:CPU_THREADS" placeholder:"CPU_THREADS" help:"Number of CPU threads"`
+	Memory              string            `arg:"--memory,env:MEMORY" placeholder:"MEMORY" help:"Memory of the vm, format 1M, 1G"`
+	TemplateLabels      []string          `arg:"--template-labels" placeholder:"KEY: VALUE KEY: VALUE" help:"Adds labels to template"`
+	TemplateAnnotations []string          `arg:"--template-annotations" placeholder:"KEY: VALUE KEY: VALUE" help:"Adds annotations to template"`
+	VMLabels            []string          `arg:"--vm-labels" placeholder:"KEY: VALUE KEY: VALUE" help:"Adds labels to VMs"`
+	VMAnnotations       []string          `arg:"--vm-annotations" placeholder:"KEY: VALUE KEY: VALUE" help:"Adds annotations to VMs"`
+	Output              output.OutputType `arg:"-o" placeholder:"FORMAT" help:"Output format. One of: yaml|json"`
+	Disks               []string          `arg:"--disks" placeholder:'{"name": "test", "cdrom": {"bus": "sata"}}' '{"name": "disk2"}' help:"VM disks in json format, replace vm disk if same name, otherwise new disk is appended"`
+	Volumes             []string          `arg:"--volumes" placeholder:'{"name": "virtiocontainerdisk", "containerDisk": {"image": "kubevirt/virtio-container-disk"}}' '{"name": "disk2"}' help:"VM volumes in json format, replace vm volume if same name, otherwise new volume is appended"`
+	Debug               bool              `arg:"--debug" help:"Sets DEBUG log level"`
+
+	templateLabels      map[string]string
+	templateAnnotations map[string]string
+	vmLabels            map[string]string
+	vmAnnotations       map[string]string
+	disks               []kubevirtv1.Disk
+	volumes             []kubevirtv1.Volume
+}
+
+func (c *CLIOptions) GetDebugLevel() zapcore.Level {
+	if c.Debug {
+		return zapcore.DebugLevel
+	}
+	return zapcore.InfoLevel
+}
+
+func (c *CLIOptions) GetCPUSockets() uint32 {
+	res, _ := strconv.ParseUint(c.CPUSockets, 10, 32)
+	return uint32(res)
+}
+
+func (c *CLIOptions) GetCPUCores() uint32 {
+	res, _ := strconv.ParseUint(c.CPUCores, 0, 32)
```

Comment:
would be good to update the other 2 as well :)

Reply:
still relevant

Reply:
Right, I forgot on them